### ISSUE TITLE
Element.executeJs() silently swallows any JavaScript errors

### ISSUE THREAD

**Issue by mvysny** [vaadin/flow#17485]
### Description of the bug

Calling `getElement().executeJs("someFunnyFunctionThatDoesntExist()")` from server-side will silently fail - the JS ReferenceError is logged in server-side nor client-side.

I can see that the information about the failure is sent to the server-side, since the XHR message contains the following:
```
Sending xhr message to server: {"csrfToken":"456710bb-4724-4d05-8686-97f674488d18","rpc":[{"type":"channel","node":6,"channel":23,"args":["ReferenceError: someFunnyFunctionThatDoesntExist is not defined"]}],"syncId":24,"clientId":24} [FlowClient-d5d5e377.js:1:8091](http://localhost:8080/VAADIN/build/FlowClient-d5d5e377.js)
```
However, nothing is logged on the server-side.

### Expected behavior

Something should be logged as WARN/ERROR, either in client-side or in server-side. Even better, an exception could be thrown in the server-side.

### Minimal reproducible example

```java
@Route("")
public class MainView extends VerticalLayout {

    public MainView() {
        add(new Button("Press me", e -> {
            getElement().executeJs("someFunnyFunctionThatDoesntExist()")
                    .then((SerializableConsumer<JsonValue>) jsonValue -> Notification.show("Never gets called"));
        }));
    }
}
```

### Versions

- Vaadin / Flow version: 24.1.5
- Java version: 17

**Comment by Legioth** [comment]
That's because you're (accidentally) explicitly suppressing the error.

If you don't have any `.then` on the execution, then there will be an error message in the browser console.

<img width="844" alt="Screenshot 2023-08-23 at 12 54 54" src="https://github.com/vaadin/flow/assets/1294657/becddb5d-ea05-4add-9c6f-f2c27335bdc3">

If you have a `.then`, then the exception will be passed to the server and delivered to the second optional parameter of `.then`. If no such parameter is passed, then the error is silently ignored and that's probably something that we should look into addressing.
```java
getElement().executeJs("someFunnyFunctionThatDoesntExist()").then(
		ignore -> Notification.show("Never gets called"),
		ignore -> Notification.show("An error occurred"));
```

**Comment by Legioth** [comment]
So your example code is really akin to this native Java example that also silently ignores the exception:

```java
CompletableFuture.supplyAsync(() -> {
	throw new RuntimeException();
}).thenAccept(value -> System.out.println("This never gets called"));
```

**Comment by mshabarov** [comment]
IMO we can try add a default handler when the seconds handler (custom error handler) is not given to `executeJs`.
I don't know if it would be possible to switch between the default one and custom one.
If not, we can at lest document this behaviour in javadoc and on online docs.

**Comment by mcollovati** [comment]
Currently, if an error handler is not provided, the behavior is to log the error message at debug level (`PendingJavaScriptInvocation` category)

**Comment by mcollovati** [comment]
It would be good to document how error handling works [here](https://vaadin.com/docs/latest/advanced/browser-access#executing-javascript-in-the-browser) and [here](https://vaadin.com/docs/latest/create-ui/element-api/client-server-rpc#return-values)

**Comment by mvysny** [comment]
I'd definitely suggest to use a default handler that logs an exception stacktrace into slf4j. That will help the programmer tremendously: he can quickly find where the offending javascript is executed, and can quickly fix the code in Java.

**Comment by mcollovati** [comment]
@mvysny `PendingJavaScriptInvocation` already logs errors at debug level when there is no custom error handler.
Are you talking about a different handler? O the suggestion is to change the level from DEBUG to WARN or ERROR?

**Comment by mvysny** [comment]
Yup, exactly. By default my log level for Vaadin is set to INFO or WARN (so that my log is not cluttered), causing the errors to be not logged. Logging them as WARN or ERROR would fix things for me.
